src/libs/SearchUtils.ts

```diff
@@ -406,8 +406,34 @@ function isSearchResultsEmpty(searchResults: SearchResults) {
     return !Object.keys(searchResults?.data).some((key) => key.startsWith(ONYXKEYS.COLLECTION.TRANSACTION));
 }
 
-function getQueryHashFromString(query: SearchQueryString): number {
-    return UserUtils.hashText(query, 2 ** 32);
+function getQueryHash(query: SearchQueryJSON): number {
+    let orderedQuery = '';
+    if (query.policyID) {
+        orderedQuery += `${CONST.SEARCH.SYNTAX_ROOT_KEYS.POLICY_ID} ${query.policyID} `;
+    }
+    orderedQuery += ` ${CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE}: ${query.type} `;
+    orderedQuery += ` ${CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS}: ${query.status}`;
+    orderedQuery += ` ${CONST.SEARCH.SYNTAX_ROOT_KEYS.SORT_BY}: ${query.sortBy} `;
+    orderedQuery += ` ${CONST.SEARCH.SYNTAX_ROOT_KEYS.SORT_ORDER}: ${query.sortOrder} `;
```

Comment:
This is already listed as a bug here: https://github.com/Expensify/App/issues/51044.

Reply:
Hmm I think this should be fixed on BE, because recentSearches are generated on BE. In case when we are searching for some query twice, then it should be renewed in history and not duplicated.

Reply:
I also gave one argument for having sortOrder etc. here https://github.com/Expensify/App/pull/50921#discussion_r1806361272.

Reply:
WDYT @rayane-djouah

Reply:
We have a [separate issue](https://github.com/Expensify/App/issues/51044) for this, so let's treat it as NAB for this PR and keep the discussion in that issue.